src/presentation/api/reports/ReportsController.ts

```diff
@@ -0,0 +1,35 @@
+import { getTestTypes, getReports } from '../../../application/service';
+import { Request, Response } from 'express';
+import { Authorized, Get, JsonController, UseBefore, Res, Req, QueryParam } from 'routing-controllers';
+import { hasPermission } from '../../middleware/hasPermission';
+import { VIEW_ADMIN_REPORTS } from '../../../domain/model/authentication/Permissions';
+import { ApiError, apiErrorCodes } from '../../dtos/ApiError';
+import { transformReportTestResultsToReport } from '../../transformers/reports/transformReportTestResultsToReport';
+
+import { writeToStream } from '@fast-csv/format';
+
+@Authorized()
+@JsonController('/v1/reports')
+export class ReportsController {
+  private getReports = getReports;
+  private getTestTypes = getTestTypes;
+
+  // Only used for data reporting/export in MVP. Very heavy query as database grows.
+  @Get('/test-results')
+  @UseBefore(hasPermission(VIEW_ADMIN_REPORTS))
+  async getReportByTestTypeId(@QueryParam('testTypeId') testTypeId: string, @Req() req: Request, @Res() res: Response) {
+    const testType = await this.getTestTypes.byId(testTypeId);
+
+    if (!testType) {
+      throw new ApiError(404, apiErrorCodes.TEST_TYPE_NOT_FOUND);
+    }
+
+    const tests = await this.getReports.getTestResults(testType.id);
```

Comment:
Maybe not for now but would it make sense to add a date to the request parameters and return only results after a certain date?